### PR TITLE
kubelet/cm/cpumanager: Improving test coverage

### PR DESCRIPTION
**Pull request by tmilox**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

If applied this commit will increase the unit tests code coverage of `pkg/kubelet/cm/cpumanager`.

- Initial unit tests code coverage

Screenshot 2023-01-02 at 20 34 22

- After Changes

Screenshot 2023-01-03 at 14 07 09

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by TommyStarK** [comment]
Hello @bart0sh 

I opened this one but the robot failed to label it properly. Had to do it manually. Not sure if I should add the size myself for example or if I can trigger the label step again. 

Happy new year, best wishes. 
Cheers,

File: pkg/kubelet/cm/cpumanager/policy_options_test.go [hunk 1]
```diff
@@ -109,53 +109,75 @@ func TestPolicyOptionsAvailable(t *testing.T) {
...
 		},
 		{
-			description:   "Align by socket enabled with topology manager single numa node",
-			policyOption:  map[string]string{AlignBySocketOption: "true"},
+			description:   "Distribute CPUs accros numa nodes with invalid policy",
+			policyOption:  map[string]string{DistributeCPUsAcrossNUMAOption: "invalid"},
 			topology:      topoDualSocketMultiNumaPerSocketHT,
-			topoMgrPolicy: topologymanager.PolicySingleNumaNode,
+			topoMgrPolicy: topologymanager.PreferClosestNUMANodes,
 			expectedErr:   true,
```
Comment (bart0sh) [inline pkg/kubelet/cm/cpumanager/policy_options_test.go:138]: `expectedValidateErr` is missing here. You should either put it everywhere or only use it when it's value is `true`.

**Comment by bart0sh** [comment]
@TommyStarK 
> Not sure if I should add the size myself for example or if I can trigger the label step again.

I hope the robot will relabel this PR when it's rebased.

File: pkg/kubelet/cm/cpumanager/policy_options_test.go (hunk 1 above)
Comment (TommyStarK) [reply pkg/kubelet/cm/cpumanager/policy_options_test.go:136]: My bad, sorry. Resolved in 832cb5ad9e3

File: pkg/kubelet/cm/cpumanager/policy_options_test.go [hunk 2]
```diff
@@ -168,10 +185,13 @@ func TestValidateStaticPolicyOptions(t *testing.T) {
 			topoMgrStore := topologymanager.NewFakeManagerWithPolicy(topoMgrPolicy)
 
 			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, pkgfeatures.CPUManagerPolicyAlphaOptions, true)()
-			policyOpt, _ := NewStaticPolicyOptions(testCase.policyOption)
-			err := ValidateStaticPolicyOptions(policyOpt, testCase.topology, topoMgrStore)
+			policyOpt, err := NewStaticPolicyOptions(testCase.policyOption)
+			if err == nil && testCase.expectedErr {
```
Comment (bart0sh) [inline pkg/kubelet/cm/cpumanager/policy_options_test.go:189]: This check misses the case with ```expectedErr = false``` and ```err != nil```. Can you use the same approach as with ```expectedValidateErr``` below?

Comment (TommyStarK) [reply pkg/kubelet/cm/cpumanager/policy_options_test.go]: Sure thing, done in 1578af56b87

**Comment by TommyStarK** [comment]
@bart0sh Do you want me to squash my commits into a single one ?

**Comment by bart0sh** [comment]
@TommyStarK it would be nice, but you don't have to. `tide/merge-method-squash` should do it.
Please, read [this guide](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#squashing) for more details.

**Comment by TommyStarK** [comment]
@bart0sh It's done. Thank you as usual for taking the time and sharing this guide I obviously missed

**Comment by TommyStarK** [comment]
@bart0sh PR has been rebased

**Comment by bart0sh** [comment]
/lgtm
@klueska can you have a look?

**Comment by TommyStarK** [comment]
@derekwaynecarr I know you are super busy, another small coverage improvement

**Comment by TommyStarK** [comment]
@klueska when you have time I'd like to complete this PR and merge it

**Pull request closed without merging**